### DISCUSSION TITLE
若视频有多语字幕, 字幕翻译是怎么工作的?

### DISCUSSION THREAD

**Discussion by XMB-7** [immersive-translate/immersive-translate#2025] [Help] [answered]
### 概述 | Summary

场景一, youtube视频同时有中英人工精校字幕, 此时开启字幕翻译, 它是直接做了个拼接? 还是画蛇添足地舍弃了精校中文字幕, 先耗费token把英文字幕翻译了, 然后再做拼接?

场景二, youtube视频同时有英文精校人工字幕和中文自动翻译字幕, 此时开启字幕翻译, 跟场景一处理方式完全一样吗, 是否花费token?

场景三, youtube视频只有英文字幕(无论人工精校还是自动生成),  此时开启字幕翻译, 应该就是花费token翻译并拼接了吧? 

主要对前两个场景有疑问.

**Comment by theowenyoung** [comment, marked as answer]
> 场景一, youtube视频同时有中英人工精校字幕, 此时开启字幕翻译, 它是直接做了个拼接? 还是画蛇添足地舍弃了精校中文字幕, 先耗费token把英文字幕翻译了, 然后再做拼接?

默认是直接做拼接，可以在设置页设置优先级。

> 场景二, youtube视频同时有英文精校人工字幕和中文自动翻译字幕, 此时开启字幕翻译, 跟场景一处理方式完全一样吗, 是否花费token?

默认情况下，将使用 英文精校人工字幕+中文自动翻译字幕 拼接，依然不会花费 token.


场景三, youtube视频只有英文字幕(无论人工精校还是自动生成), 此时开启字幕翻译, 应该就是花费token翻译并拼接了吧?

默认情况下，有自动生成的英文和中文字幕，优先会用自动生成的，不会花token. 



但是你可以在设置页，改变优先级，指定让沉浸式翻译来翻译所有的字幕。
